### PR TITLE
add ncdu 1.15.1

### PR DESCRIPTION
**Pull request by jclulow**

### CODE REVIEW

**Review by oetiker — CHANGES_REQUESTED** [review]
an entryu in https://github.com/omniosorg/omnios-extra/blob/master/doc/packages.md is missing

File: build/ncdu/build.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+SUMMARY="$PROG - NCurses Disk Usage"
+DESC='Disk usage analyzer with an ncurses interface'
+
+set_arch 64
+
+set_mirror 'https://dev.yorhel.nl/download/'
+set_checksum sha256 \
+    'b02ddc4dbf1db139cc6fbbe2f54a282770380f0ca5c17089855eab52a9ea3fb0'
+
+CFLAGS+=' -I/usr/include/ncurses '
```
Comment (citrus-it) [inline build/ncdu/build.sh:33]: Does this work with CPPFLAGS instead?

File: build/ncdu/build.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+VER=1.15.1
+PKG=ooce/util/ncdu
+SUMMARY="$PROG - NCurses Disk Usage"
+DESC='Disk usage analyzer with an ncurses interface'
+
+set_arch 64
+
+set_mirror 'https://dev.yorhel.nl/download/'
+set_checksum sha256 \
+    'b02ddc4dbf1db139cc6fbbe2f54a282770380f0ca5c17089855eab52a9ea3fb0'
```
Comment (citrus-it) [inline build/ncdu/build.sh:31]: I've put the source on the omnios mirror now, so please drop these three lines

File: build/ncdu/build.sh (hunk 1 above)
Comment (citrus-it) [inline build/ncdu/build.sh:33]: Does this work with CPPFLAGS instead?

File: build/ncdu/local.mog [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# }}}
+#
+# Copyright 2021 Oxide Computer Company
+
+license COPYING license=MIT
+
+# vim:ts=4:sw=4:et:fdm=marker
```
Comment (citrus-it) [inline build/ncdu/local.mog:18]: We generally do not use the fold markers of the vim modeline in the .mog files, just a trailing blank line.

**Review by citrus-it — CHANGES_REQUESTED** [review]
Thanks Josh, that looks like a useful utility. Just a few nits, and please can you add the package to `doc/packages.md` so that our package update monitoring plugins can keep an eye on updates? 
Setting the URL there to https://dev.yorhel.nl/download should be enough for it to work. Please set the maintainer field to Oxide if they/you will be looking after it.

Can you also please add a line to `doc/baseline`?

File: build/ncdu/build.sh [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+CFLAGS+=' -I/usr/include/ncurses '
+
+init
+download_source "" "$PROG" "$VER"
+prep_build
+build
+strip_install
```
Comment (citrus-it) [inline build/ncdu/build.sh:39]: Since CTF conversion seems to work on all OmniOS releases, can you replace these two lines?

```suggestion
build -ctf
```

**Comment by jclulow** [comment]
With my apologies for it taking so long, I think I've addressed all the feedback!  I've confirmed that the package still builds.  Let me know if you need anything else.

**Review by hadfl — APPROVED** [review]

**Pull request merged**